mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -940,49 +940,114 @@ mlir::scf::tileAndFuseProducerOfSlice(
-  Value tiledAndFusedProducer = fusedProducerInfo.tiledAndFusedProducer;
-  FailureOr<Value> initValue = tensor::getOrCreateDestination(
-      rewriter, fusableProducer.getOwner()->getLoc(), fusableProducer);
-  if (succeeded(initValue)) {
-
-    YieldTiledValuesFn newYieldValuesFn =
-        [&](RewriterBase &innerRewriter, Location loc, ValueRange /*ivs*/,
-            ValueRange newRegionIterArgs, SmallVector<Value> &tiledResult,
-            SmallVector<SmallVector<OpFoldResult>> &tiledOffset,
-            SmallVector<SmallVector<OpFoldResult>> &tiledSizes)
-        -> LogicalResult {
-      OpBuilder::InsertionGuard g(innerRewriter);
-      if (auto tiledDestStyleOp =
-              tiledAndFusedProducer
-                  .getDefiningOp<DestinationStyleOpInterface>()) {
-        rewriter.setInsertionPoint(tiledDestStyleOp);
-        Value newRegionArg = newRegionIterArgs.back();
+  Operation *originalOwner = fusedProducerInfo.origProducer.getOwner(),
+            *tiledOwner = fusedProducerInfo.tiledOps[0];
+
+  Location loc = originalOwner->getLoc();
+  // a. collect all init Value to be appended
+  SmallVector<unsigned> initNumberList =
+      yieldResultNumber.empty() ? llvm::to_vector(llvm::seq<unsigned>(
+                                      0, originalOwner->getNumResults()))
+                                : llvm::to_vector(yieldResultNumber);
+  SmallVector<Value> initValueList;
+  for (const auto &resultNumber : initNumberList) {
+    FailureOr<Value> initValue = tensor::getOrCreateDestination(
+        rewriter, loc, originalOwner->getResult(resultNumber));
+    if (succeeded(initValue)) {
+      initValueList.push_back(initValue.value());
+    } else {
+      return failure();
+    }
+  }
+
+  YieldTiledValuesFn newYieldValuesFn =
+      [&](RewriterBase &innerRewriter, Location loc, ValueRange /*ivs*/,
+          ValueRange newRegionIterArgs, SmallVector<Value> &tiledResult,
+          SmallVector<SmallVector<OpFoldResult>> &tiledOffset,
+          SmallVector<SmallVector<OpFoldResult>> &tiledSizes) -> LogicalResult {
+    OpBuilder::InsertionGuard g(innerRewriter);
+
+    // get sliceOp tile information
+    SmallVector<OpFoldResult> sliceOffset = sliceOp.getMixedOffsets(),
+                              sliceSizes = sliceOp.getMixedSizes();
+
+    // expect all strides of sliceOp being 1
+    if (llvm::any_of(sliceOp.getMixedStrides(), [](OpFoldResult ofr) {
+          return !isConstantIntValue(ofr, 1);
+        }))
+      return failure();
+
+    unsigned sliceResultNumber =
+        fusedProducerInfo.origProducer.getResultNumber();
+
+    auto tilableOp = cast<TilingInterface>(originalOwner);
+    // b. get iterDomain Offset and Sizes based on sliceOp tile
+    SmallVector<OpFoldResult> iterDomainOffset, iterDomainSizes;
+    // skip tensor.pack/unpack/pad, which expects single opResult
+    if (tilableOp->getNumResults() > 1 &&
+        failed(tilableOp.getIterationDomainTileFromResultTile(
```

Comment:
It might be better to just not yield the result if you cannot compute the iteration domain tile from result tile. I am fine with this being a failure, but I think that in terms of generality, not having an op implement this method would just mean you cannot yield the results, but can still fuse with it.

Reply:
Yes, I agree with you in terms of generality. But note that, here is the callback of `addInitOperandsToLoopNest`, If we could not compute the iteration domain tile from result tile, it rightly means failure of adding `InitOperands` to nest loop. Otherwise, it is ambiguous to define what is success or failed. BTW, I find that it would also return failure in [master](https://github.com/llvm/llvm-project/blob/main/mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp#L1077) for your case. To reach your expectation, I think we may modify there to decouple the yield result with fuse or not, Please let me known your thought or just keep current status.

Reply:
Its not really a failure... you can have an implementation where you dont add any init operands cause you cannot find how to reconstruct the result tensor. Basically `failure` is for "halt the compilation cause something went wrong". So I would not return failure, but continue. Doesnt the code you point to also "continue".

Reply:
> but continue. Doesnt the code you point to also "continue".

Sorry for this missing code line due to recent master change. I just attach concrete code below for you reference:

```
if (failed(yieldReplacementForFusedProducer(
        rewriter, candidateSliceOp, fusedResult.value(), loops))) {
  return rewriter.notifyMatchFailure(
      fusableProducer.getOwner(), "failed to replacement value for this "
                                  "oepration from within the tiled loop");
}
```

Reply:
Correct me if I am wrong in my head about implementation details, but if you just continue here, it would mean that you arent yielding the value, and then the `addInit*` wont add an init for that value. So it would still work. There is no reason to actually raise an error here..

Reply:
> There is no reason to actually raise an error here..

Yes, but what I attached is current status in master... Did you mean we should also change it?

Reply:
> It might be better to just not yield the result if you cannot compute the iteration domain tile from result tile (instead of it being a failure).

Wait, wait... Seems that I misunderstood before. Correct me if I am wrong, did you mean we should just return `success` for this case but yield nothing and add no init. Then we can still fuse them?

Reply:
I can give more implementation details to show the difficult on doing that. The callback `yieldTiledValuesFn` is called in [yieldTiledValuesAndReplaceLoop](https://github.com/Yun-Fly/llvm-project/blob/yunfei/fuse_yield_multiouts/mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp#L360), where the `InitOperand` has already been appended. As the result, we must raise error here...

Reply:
Since there are no new replies for a week, hopefully we have reached agreements for this. If not, I can address it post landing.

Reply:
Sorry, fell of my radar... let me get the state back on this.

Reply:
> I can give more implementation details to show the difficult on doing that. The callback `yieldTiledValuesFn` is called in [yieldTiledValuesAndReplaceLoop](https://github.com/Yun-Fly/llvm-project/blob/yunfei/fuse_yield_multiouts/mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp#L360), where the `InitOperand` has already been appended. As the result, we must raise error here...

I see your point now. It will take more refactoring to make sure the init operands are added consistently so that we can handle this. Can you just add a comment here as to why this is a failure for now and capture the discussion we had here in the comment (also point to this thread of conversation if possible)

It looks good to me to land.

Reply:
> Can you just add a comment here as to why this is a failure for now

Yes, sure. It is useful as a kind reminder for all of us.

> It looks good to me to land.

Thanks again!